src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs

```diff
@@ -624,13 +624,13 @@ public async Task ReadISetTOfHashSetT()
 
             if (result.First().Contains(1))
             {
-                Assert.Equal(new HashSet<int> { 1, 2 }, result.First());
```

Comment:
Apparently the xunit comparer for sets was using unsupported reflection, so I changed it to something equivalent.